libs/langchain/langchain/output_parsers/openai_functions.py

```diff
@@ -28,6 +28,8 @@ def parse_result(self, result: List[Generation]) -> Any:
             func_call = message.additional_kwargs["function_call"]
         except ValueError as exc:
             raise OutputParserException(f"Could not parse function call: {exc}")
+        except KeyError as key:
```

Comment:
The `ValueError` should probably be a `KeyError` since there doesn't seem to be a way to trigger a ValueError from that line. Does that seem right?

Reply:
Yep updated the PR. Only KeyError handling is required here.